runner/push_guard.py

```
@@ -0,0 +1,146 @@
+        result["push_skip_reason"] = "allow_push_not_set"
+        return result
+    normalized = str(final_outcome or "").strip().lower()
+    if normalized != "success":
+        result["push_skip_reason"] = f"final_outcome={normalized or 'unknown'}_is_not_success"
+        return result
+    if not workdir:
+        result["push_skip_reason"] = "no_workdir"
+        return result
+    result["push_attempted"] = "1"
+    try:
+        proc = subprocess.run(
```

Ruff (0.16.3)

[error] `subprocess` call: check for execution of untrusted input

(S603)